pkg/secretsencrypt/controller.go

```diff
@@ -41,11 +42,21 @@ type handler struct {
 
 func Register(
 	ctx context.Context,
-	k8s kubernetes.Interface,
+	controllerName string,
 	controlConfig *config.Control,
 	nodes coreclient.NodeController,
 	secrets coreclient.SecretController,
 ) error {
+	cfg := controlConfig.Runtime.KubeConfigAPIServer
```

Comment:
Why are we replacing the built-in controller kubernetes client with one that uses the apiserver's kubeconfig? If we need to make our own client for some reason, we should use:
```suggestion
	cfg := controlConfig.Runtime.KubeConfigKubeConfigSupervisor
```

Reply:
This initialization of this controller has been moved to where a I no longer have access to the cfg to pass as a argument into the Register function. I wasn't 100% sure which KubeConfig to use, so I went with the APIServer one because the this controller is now directly tied to the existence of the apiserver.

Reply:
I don't actually use the k8s interface for anything in the controller, its just used to initalize the EventRecorder.